test/Web/ScottySpec.hs

```diff
@@ -307,6 +320,35 @@ spec = do
         it "responds with 200 OK if the parameter is not found" $ do
           get "/b/potato" `shouldRespondWith` 200
 
+    describe "files" $ do
+      withApp (Scotty.post "/files" $ do
+                  fs <- files -- XXX Control.Monad.Trans.Resource.register': The mutable state is being accessed after cleanup. Please contact the maintainers.
```

Comment:
the current implementation of `files` currently throws this use-after-free exception due to how it internally uses `register'` from resourcet.

Reply:
fixed